java/Makefile

```
@@ -23,6 +23,17 @@ JAVAC ?= javac
 JAVADOC ?= javadoc
 JAR ?= jar
 
+ifeq ($(shell uname), Darwin)
+  JAVA_HOME ?= $(/usr/libexec/java_home)
+else
+ifeq ($(shell which javac > /dev/null 2>&1 && echo found || echo nonexistent), found)
+  JAVA_HOME ?= $(shell readlink -f $(shell which javac) | sed "s:/bin/javac::")
+else
+  $(info Failure: Here is the output of: which javac)
```

checkmake (0.3.2)

[warning] Target "$(info Failure" should be declared PHONY.

(phonydeclared)